src/v/http/request_builder.h

```diff
@@ -71,10 +72,28 @@ class request_builder {
     // Adds a Bearer auth token header
     request_builder& with_bearer_auth(std::string_view token);
 
+    // Set a shared credentials applier for request signing, meant to
+    // be used for signing requests with AWS SigV4. If the request
+    // has a payload (POST/PUT) and the endpoint requires the payload
+    // hash be included in the signature, use `with_payload_hash` to
+    // add the proper header that will be used in signing.
+    request_builder& with_apply_credentials(
+      ss::lw_shared_ptr<cloud_roles::apply_credentials> apply_creds);
```

Comment:
Why does our http client have a bunch of aws specific logic in it? Is there a reason this can't go in a layer above this one or some wrapper around our client?